privacyidea/models.py

```diff
@@ -1050,9 +1052,69 @@ def __init__(self, resolver_id=None, realm_id=None,
                                  .first().id
 
 
+class TokenOwner(MethodsMixin, db.Model):
+    """
+    This tables stores the owner of a token.
+    A token can be assigned to several users.
+    """
+    __tablename__ = 'tokenowner'
+    id = db.Column(db.Integer(), Sequence("tokenowner_seq"), primary_key=True,
+                   nullable=True)
+    token_id = db.Column(db.Integer(), db.ForeignKey('token.id'))
+    token = db.relationship('Token', lazy='joined', backref='token_list')
+    resolver = db.Column(db.Unicode(120), default=u'', index=True)
+    resolver_type = db.Column(db.Unicode(120), default=u'')
+    user_id = db.Column(db.Unicode(320), default=u'', index=True)
+    realm_id = db.Column(db.Integer(), db.ForeignKey('realm.id'))
+    realm = db.relationship('Realm', lazy='joined', backref='realm_list')
+
+    def __init__(self, token_id=0, serial=None, user_id=None, resolver=None, resolver_type=None, realm_id=0, realmname=None):
+        """
+        Create a new token assignment to a user.
+
+        :param token_id: The database ID of the token
+        :param serial:  The alternate serial number of the token
+        :param resolver: The identifying name of the resolver
+        :param resolver_type: The type of the resolver
+        :param realm_id: The database ID of the realm
+        :param realmname: The alternate name of realm
+        """
+        if realm_id:
+            self.realm_id = realm_id
+        elif realmname:
+            r = Realm.query.filter_by(name=realmname).first()
+            self.realm_id = r.id
+        if token_id:
```

Comment:
Are there databases for which the ID of the first generated token would be ``0``? If yes, we should change the default argument value to ``None`` and check ``token_id is not None`` here.

Reply:
You are right.

Will change this.

Reply:
done in 0d1b013f